libcontainer/cgroups/devices/devices_emulator.go

```diff
@@ -79,19 +78,24 @@ func (e *Emulator) IsAllowAll() bool {
 	return e.IsBlacklist() && len(e.rules) == 0
 }
 
-var devicesListRegexp = regexp.MustCompile(`^([abc])\s+(\d+|\*):(\d+|\*)\s+([rwm]+)$`)
-
 func parseLine(line string) (*deviceRule, error) {
-	matches := devicesListRegexp.FindStringSubmatch(line)
-	if matches == nil {
-		return nil, errors.New("line doesn't match devices.list format")
+	// The three fields are: node major:minor perms.
+	fields := strings.Fields(line)
+	if len(fields) != 3 {
+		return nil, fmt.Errorf("malformed device.list rule %s", line)
+	}
+	// Parse the major:minor pair.
+	mm := strings.Split(fields[1], ":")
```

Comment:
So, I think since this is not user-provided input, so we don't have to be extra suspicious when parsing it.

`SplitN` is nice as it prevents bad input attacks (e.g. something like a few million `:` characters), but here we do not ever expect it, and so I lean towards better code readability (which SplitN decreases a bit).

Reply:
> So, I think since this is not user-provided input, so we don't have to be extra suspicious when parsing it.

Ack, yes, definitely not a blocker. I was also looking at the code, and specifying a limit also skips a `bytes.Count()` to be done; https://github.com/golang/go/blob/2ebe77a2fda1ee9ff6fd9a3e08933ad1ebaea039/src/strings/strings.go#L243-L245, but agreed that this is not user input, so malformed / bad values _should_ not be a thing here.